strawberryfields/engine.py

```diff
@@ -336,8 +360,144 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             # empty sequence
             pass
         else:
-            result.state = self.backend.state(modes, **state_options)  # tfbackend.state can use kwargs
+            result.state = self.backend.state(
+                modes, **state_options
+            )  # tfbackend.state can use kwargs
         return result
 
 
+class StarshipEngine(BaseEngine):
+    """
+    Starship quantum program executor engine.
+
+    Executes :class:`.Program` instances on the chosen remote backend, and makes
+    the results available via :class:`.Result`.
+
+    Args:
+        backend (str, BaseBackend): name of the backend, or a pre-constructed backend instance
+    """
+
+    API_DEFAULT_REFRESH_SECONDS = 1
+    HARDWARE_BACKENDS = ("chip0",)
+
+    def __init__(self, api_client_params=None):
+        # Only chip0 backend supported initially.
+        backend = "chip0"
+        super().__init__(backend)
+
+        api_client_params = api_client_params or {}
+        self.client = APIClient(**api_client_params)
+        self.jobs = []
+
+    def __str__(self):
+        return self.__class__.__name__ + "({})".format(self.backend_name)
+
+    def reset(self, backend_options=None):
+        """
+        Reset must be called in order to submit a new job. This clears the job queue as well as
+        any ran Programs.
+        """
+        if backend_options is None:
+            backend_options = {}
+
+        super().reset(backend_options)
+        self.jobs.clear()
+
+    def _init_backend(self, *args):
+        """
+        TODO: This does not do anything right now.
+        """
+        # Do nothing for now...
+        pass
+
+    def generate_job_content(self, name, shots, blackbird_code):
+        """
+        Generates a string representing the Blackbird code that will be sent to the server.
+        Assumes the current backend as the target.
+
+        Args:
+            name (str): The name of the job to be created (e.g. StateTeleportation).
+            shots (int): The number of shots.
+            blackbird_code: The blackbird code of the job.
+
+        Returns:
+            str: A string containing the job content to be sent to the server.
+        """
+        target = self.backend_name
+        return "\n".join(
+            [
+                "name {name}",
+                "version 1.0",
+                "target {target} (shots={shots})",
+                "",
+                "{blackbird_code}",
+            ]
+        ).format(name=name, target=target, shots=str(shots), blackbird_code=blackbird_code)
+
+    def _queue_job(self, job_content):
+        """
+        Create a Job instance based on job_content, and send the job to the API. Append to list
+        of jobs.
+
+        Args:
+            job_content (str): The Blackbird code to execute
```

Comment:
```suggestion
            job_content (str): the Blackbird code to execute
```
(Aside - we should work out if Blackbird should be capitalized or not, I think I've seen both across the codebase).